src/main/java/org/springframework/integration/kafka/dsl/KafkaProducerMessageHandlerSpec.java

```diff
@@ -195,6 +195,51 @@
 	}
 
 	/**
+	 * Configure a SpEL expression to determine the timestamp at runtime against a
+	 * request Message as a root object of evaluation context.
+	 *
+	 * @param timestampExpression the timestamp expression to use.
+	 * @return the spec.
+	 *
+	 * @since 3.0
+	 */
+	public KafkaProducerMessageHandlerSpec<K, V> timestampExpression(String timestampExpression) {
+		return this.timestampExpression(PARSER.parseExpression(timestampExpression));
+	}
+
+	/**
+	 * Configure a {@link Function} that will be invoked at run time to determine the Kafka record timestamp
+	 * will be stored in the topic. Typically used with a Java 8 Lambda expression:
+	 * <pre class="code">
+	 * {@code
+	 * .timestamp(m -> m.getHeaders().get("mytimestamp_header", Long.class))
+	 * }
+	 * </pre>
+	 * @param timestampFunction the partitionId function.
+	 * @param <P> the expected payload type.
+	 * @return the spec.
+	 *
+	 * @since 3.0
```

Comment:
Doh. We don't need `@since` here. This class has been fully introduced exactly `@since 3.0` 😄